File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/DTOs/RegisterClientUserDTO.cs
using System.ComponentModel.DataAnnotations;

namespace BankingPaymentsApp_API.DTOs
{
    public class RegisterClientUserDTO
    {
        [Required]
        public string UserFullName { get; set; }
        [Required]
        public string UserName { get; set; }
        [Required]
        public string UserEmail { get; set; }
        [Required]
        public string UserPhone { get; set; }
        [Required]
        public int UserRoleId { get; set; } = 3;
        [Required]
        public int BankId { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        public string ConfirmPassword { get; set; }
        [Required]
        [DataType(DataType.Date)]
        public DateTime DateOfBirth { get; set; }
        [Required]
        public string Address { get; set; }
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/DTO/RegisterClientUserDTO.ts
export interface RegisterClientUserDTO {
  userFullName: string;
  userName: string;
  userEmail: string;
  userPhone: string;
  userRoleId: number;
  bankId:number;
  password: string;
  confirmPassword: string;
  dateOfBirth: string; 
  address: string;
  accountId?: number;  
}
